hummingbot/connector/exchange/ndax/ndax_order_book_message.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+
+from collections import namedtuple
+from typing import (
+    Dict,
+    List,
+    Optional,
+)
+
+from hummingbot.core.data_type.order_book_row import OrderBookRow
+from hummingbot.core.data_type.order_book_message import (
+    OrderBookMessage,
+    OrderBookMessageType,
+)
+
+NdaxOrderBookEntry = namedtuple("NdaxOrderBookEntry", "mdUpdateId accountId actionDateTime actionType lastTradePrice orderId price productPairCode quantity side")
+NdaxTradeEntry = namedtuple("NdaxTradeEntry", "tradeId productPairCode quantity price order1 order2 tradeTime direction takerSide blockTrade orderClientId")
+
+
+class NdaxOrderBookMessage(OrderBookMessage):
+    def __new__(
```

Comment:
It might be better to implement the validation in __init__ rather than implementing __new__

Reply:
My understanding of the usage of `__new__` is that it is generally used when there is a need to include some logic prior to the initialisation of teh object. Also I am using the base class as reference